packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -81,4 +82,52 @@ export default class Pools {
       await _xo.mergeXenPools(source._xapiId, target._xapiId, force)
     }
   }
+
+  async listPoolsMatchingCriteria({
+    hostVersion,
+    minAvailableHostMemory = 0,
+    minAvailableSrSize = 0,
+    minHostCpus = 0,
+    poolName,
+    srName,
+  }) {
+    const hostsByPool = {}
+    const srsByPool = {}
+    const pools = []
+    for (const obj of this._xo.objects.values()) {
+      if (obj.type === 'host') {
+        if (hostsByPool[obj.$pool] === undefined) {
+          hostsByPool[obj.$pool] = []
+        }
+        hostsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'SR') {
+        if (srsByPool[obj.$pool] === undefined) {
+          srsByPool[obj.$pool] = []
+        }
+        srsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'pool') {
+        pools.push(obj)
+      } else {
+        continue
+      }
+    }
+
+    return pools.filter(
+      pool =>
+        (poolName === undefined || new RegExp(poolName).test(pool.name_label)) &&
```

Comment:
Not sure it's a good idea to make it a regexp.

At least it should be explicit in the param name, otherwise it might be surprise.